### PR TITLE
[e2e-tests]VMIs in tests require vmx feature to run

### PR DESCRIPTION
**Pull request by Barakmor1**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lyarwood** [comment]
https://www.qemu.org/docs/master/system/i386/hyperv.html#existing-enlightenments

Sets out that this is Intel only (TIL):

```
The enlightenment is nested specific, it targets Hyper-V on KVM guests. When enabled, it provides Enlightened VMCS version 1 feature to the guest. The feature implements paravirtualized protocol between L0 (KVM) and L1 (Hyper-V) hypervisors making L2 exits to the hypervisor faster. The feature is Intel-only.

Note: some virtualization features (e.g. Posted Interrupts) are disabled when hv-evmcs is enabled. It may make sense to measure your nested workload with and without the feature to find out if enabling it is beneficial.

Requires: hv-vapic
```

LGTM but can you add some context to the PR description and commit?

**Comment by Barakmor1** [comment]
> https://www.qemu.org/docs/master/system/i386/hyperv.html#existing-enlightenments
> 
> Sets out that this is Intel only (TIL):
> 
> ```
> The enlightenment is nested specific, it targets Hyper-V on KVM guests. When enabled, it provides Enlightened VMCS version 1 feature to the guest. The feature implements paravirtualized protocol between L0 (KVM) and L1 (Hyper-V) hypervisors making L2 exits to the hypervisor faster. The feature is Intel-only.
> 
> Note: some virtualization features (e.g. Posted Interrupts) are disabled when hv-evmcs is enabled. It may make sense to measure your nested workload with and without the feature to find out if enabling it is beneficial.
> 
> Requires: hv-vapic
> ```
> 
> LGTM but can you add some context to the PR description and commit?

Sure Done.

**Comment by xpivarc** [comment]
/hold
Adding a label for one specific test is not a scalable solution. Furthermore, there is no need for this afaik for now. Let's see if anybody else needs this.

**Comment by xpivarc** [comment]
> > /hold Adding a label for one specific test is not a scalable solution. Furthermore, there is no need for this afaik for now. Let's see if anybody else needs this.
> 
> We have created a test that assumes the vmx feature is supported by our cluster. It's important to inform users about this requirement to prevent any unexpected surprises. However, addressing this issue in a scalable way poses a challenge because each test has unique requirements, and consolidating labels might lead to unnecessary test skips.

I am afraid nobody will use this. I would prefer if the user of the test suite would rather skip EVMCS-related tests. Are we not able to use skip with labels together?

To be clear I am all for specifying what we need(hw-related or software integrations) so users have the power to skip tests that are not able to run on their clusters but I would not do it for every single little piece. Rather I would suggest a general escape hatch.

For example, in this case, a user should know that EVMCS requires Intel hardware.

**Comment by Barakmor1** [comment]
> > > /hold Adding a label for one specific test is not a scalable solution. Furthermore, there is no need for this afaik for now. Let's see if anybody else needs this.
> > 
> > 
> > We have created a test that assumes the vmx feature is supported by our cluster. It's important to inform users about this requirement to prevent any unexpected surprises. However, addressing this issue in a scalable way poses a challenge because each test has unique requirements, and consolidating labels might lead to unnecessary test skips.
> 
> I am afraid nobody will use this. I would prefer if the user of the test suite would rather skip EVMCS-related tests. Are we not able to use skip with labels together?
> 
> To be clear I am all for specifying what we need(hw-related or software integrations) so users have the power to skip tests that are not able to run on their clusters but I would not do it for every single little piece. Rather I would suggest a general escape hatch.
> 
> For example, in this case, a user should know that EVMCS requires Intel hardware.

The decision to skip certain tests is not solely based on which features the nodes support, but also on the specific workloads being tested. For example, tests involving migration may require different features than tests that do not involve migration. Aggregating labels can become complex, and some tests may require only vmx. Therefore, it may be beneficial to add vmx as a label to simplify the test selection process.

If you have any suggestions on how we can use a single label for multiple tests to improve clarity and simplicity, I would be happy to review them.

**Comment by iholder101** [comment]
> I am afraid nobody will use this. I would prefer if the user of the test suite would rather skip EVMCS-related tests. Are we not able to use skip with labels together?
> 
> To be clear I am all for specifying what we need(hw-related or software integrations) so users have the power to skip tests that are not able to run on their clusters but I would not do it for every single little piece. Rather I would suggest a general escape hatch.
> 
> For example, in this case, a user should know that EVMCS requires Intel hardware.

Hey @xpivarc,
For the record, I'd like to add my thoughts here.

Firstly, with these tests it's not enough to use a "one for all" label. Some tests with hyperV need tsc frequencies exposed, some don't. Same goes for tests running VMs with invtsc. I don't see how you can use one label in order to conclude which tests you need to skip in which scenarios.

Same goes for something like Intel hardware. Even with Intel, sometimes tsc frequencies are exposed and sometimes not. hyperv / invtsc can be supported or not supported.

We also can't assume that we have a well-defined sets of nodes, e.g. either Intel with tsc-frequencies that supports X, Y and Z cpu features, or AMD without tsc-frequencies that support A B and C cpu features. It makes total sense to me to test Kubevirt with different and heterogeneous sets of nodes. Ideally we would be able to run the test suite in every environment while skipping the right tests.

Basically, there's a tension regarding granularity. My opinion is that we should have the highest granularity we can get to make things easy and scalable, **as long we provide all of the information needed to know which tests to skip**.

Maybe in the future we could group these labels in a way that makes sense, maybe by things like CPU features,  information exposed to nodes, etc.

**Comment by iholder101** [comment]
> Right, it is not easy to do it "one for all". Generally, my thought is that if you want to test Hyperv it doesn't really make sense to run a few tests there and a few there.

Why not? :)
You just grouped hyperv tests as if they're all related, but it's not necessarily the case. For example, I might run tests on a single-node cluster, then migration tests must be skipped, but I still might want to run all of the hyperV tests (since a lot of them don't depend on migrations at all). And if we don't migrate hyperv, we don't need frequencies exposed for it at all.

> But again to circle back, I don't necessarily support any label for one or two tests.

I think this is being done incrementally and that's a good thing. We don't necessarily want to change all of the tests at once. But I think it would be great if every test would be expected to consist a list of easy-to-use-and-understand labels that would reflect the tests' dependencies.

> This to me seems like overhead and I would rather explore if we can do label filtering and text filtering at the same time.

I completely agree, we should explore and try to find easier ways to do so. This is one of the reasons why incremental changes are a great thing. As I've wrote before, things should be as easy as possible as long as we provide the information of which tests need to be skipped. IMHO something like "we don't have TSC frequencies exposed so we can skip all hyperv tests" is not good enough and would lead to more confusion.
